[16.0][MIG] project_api_client

florian-dacosta opened the pull request:
@bealdav @sebastienbeau 

J'ai essayé de me pencher un peu sur le sujet, voici une première version de la migration.
Quelques points : 
1) Maintenant qu'on ne peut plus mettre de groupe au niveau des vues, est-ce qu'il y a un moyen simple de mettre un champs en readonly ou pas selon un groupe ? 
J'ai pas en tête de solution simple pour migrer cela : https://github.com/akretion/support/blob/14.0/project_api_client/views/project_view.xml#L150
Une possibilité pas ouf serait d'ajouter un champs, du genre "is_manager" sur les external task, et de le remplir dans la surcharge du read dans le cas où le user est manager, pis de mettre les champs en readonly selon un domaine contenant ce champs "is_manager". Mais je me dis qu'il doit y avoir plus simple ?

2) @sebastienbeau J'ai découvert un peu plus la gestion des mails et ça me semble quand même chaud/moyen de maintenir cela sur le long terme, car on détourne vraiment tout le système. Déjà le fait d'avoir le chatter sans hérité de mail.thread a un certain nombre d'effet de bord. Il y a peut être plus propre que ce que j'ai fait en tout cas, j'ai juste réglé les soucis que je rencontrais au plus simple sans trop me poser de question.

3) @bealdav Est-ce que c'est uniquement cette gestion des mails/messages qui te posait problème où il y avait d'autres choses ?
J'ai pas fais plus de tests que ça, et il y a peut être encore des soucis dans le module, je ne sais pas.

florian-dacosta committed:
[16.0][MIG] project_api_client

PaulGoubert commented:
Merci @florian-dacosta d'avoir attaquer ce boulot. 
Je viens de tester sur un CI de Anothercorp en 16.
j'ai bien réussi à connecter le module support à notre ERP. 

J'ai créer une tache côté client, depuis un SO.    Lé tâche ce créer bien dans l'ERP AK.   Je recoit bien le mail de demande de support.     Par contre lorsque je clique sur le lien dans la tâches, coté AK, pour accèder au SO sur lequel la tâche à était créer j'ai ce message. 
<img width="2240" alt="Capture d’écran 2023-06-28 à 11 26 40" src="https://github.com/akretion/support/assets/99026695/9333d53a-7920-4069-b6ef-d77d6d0766c9">

Sinon j'ai remarqué que la liste des etapes n'était pas la même côté Client que Côté AK. 
<img width="2168" alt="Capture d’écran 2023-06-28 à 11 36 28" src="https://github.com/akretion/support/assets/99026695/68809d70-1058-447a-bbf8-e5eb262452fc">
<img width="2051" alt="Capture d’écran 2023-06-28 à 11 36 22" src="https://github.com/akretion/support/assets/99026695/8660ee1e-7533-448b-8e40-61a2d050bc64">

j'ai rien remarqué d'autre sinon. 
Merci

florian-dacosta commented:
Je me demande si ce soucis n'est pas déjà existant sur les versions antérieures simplement il doit être assez rare pour qu'o ne l'ait pas vu.

Ici : https://github.com/akretion/support/blob/16.0-mig-project_api_client/project_api_client/models/project.py#L329
On essaie de récupérer l'action depuis laquelle l'utilisateur crée le ticket, afin de ré-utiliser la même depuis l'url qu'on stocke sur le ticket.
Si on ne trouve pas l'action dans le contexte, on prend la première action qui vient.

@bealdav tu as fais des changements récemments sur le 14 : https://github.com/akretion/support/pull/41
Sais-tu dans quelles situations on a cette clé "params" dans le contexte, sur laquelle on se base pour récupérer l'action ?
J'ai fais 2/3 tests sur une v14 où je ne l'ai pas dans le contexte, et pareil sur une v16. Du coup, systématiquement, Odoo cherche la première action du modèle.

=> Le problème remonté par Paul est que Odoo trouve une action des sale.order au pif, qui contient un contexte mettant un filtre par défaut sur les équipes commerciale avec un active_id.
Le contexte de l'action qu'odoo trouve contient :` 'search_default_team_id': [active_id]` or active_id dans notre cas c'est l'identifiant de la sale order, ce qui fait qu'il cherche une équipe commerciale avec un identifiant de sale order et forcément il ne trouve pas.

En relisant j'ai trouvé le commentaire qui explique qui semble expliquer pourquoi on veut passer l'action dans l'url : 
"'action' or 'menu_id' key prevent redirection fails when website module is installed"
=> est-ce que cette clé param dans le contexte est aussi mise en le site web est installé ?

J'aurais tendance à dire : on récupère l'action du contexte, et si on la trouve pas, on construit l'url sans spécifier l'action.
=> Reste à savoir, est-ce que cela plantera encore avec le website installé et dans quels cas on a ou pas cette clé params dans le contexte.
@bealdav un avis ?

florian-dacosta commented:
Actually, maybe there is a better solution. It seems that we always have the action_id information from the request object. So I guess we could do it this way, I'll propose a fix

florian-dacosta committed:
[FIX] Avoid searching a default action
We could get unlucky and getting one that does not work because it need some active_id info for another object

florian-dacosta committed:
[FIX] In case description of ticket is empty, send a string instead of False to avoid webservice failure

florian-dacosta commented:
Bon, du coup j'ai retiré le fait d'ajouter une action par défaut dans l'url.
J'ai laissé le code qui essaie de récupérer l'ancienne action du contexte mais je ne sais pas si il est vraiment utile.
Il est donc possible que cette partie d'url de redirection ne fonctionne pas quand le module website est installé => A tester/corriger quand on aura le cas...

Du coup, en admettant que le bug remonté par @PaulGoubert soit corrigé, le problème restant est sur la vue des tâche par le client (voir point 1. du premier message de la PR)

bealdav commented:
En fait pour ce point je crois que @bguillot sait de quoi il s'agit, une histoire de m2m a réinitialiser je crois.
Je me rappel plus comment benoit avait fait.

florian-dacosta committed:
[FIX] form view : make field editable for manager + search view, add default filter on tag and project and make project always visible for manager

florian-dacosta committed:
Re-activate CI

florian-dacosta commented:
Cette PR est à peu près prête maintenant, point de vue module Odoo.
=> Les tests de Paul semblent concluants, et j'ai pas vu de soucis particulier.
Par contre, je n'ai pas été analysé en profondeur le code pour voir si il y avait du code mort, des fonctionalités disparues, etc, donc ce n'est forcément parfait.

Il reste aussi la CI qui est cassée, mais étant donné que c'est assez spécifique, je préfèrerai amplement que @sebastienbeau se penche dessus pour la remettre d'aplomb.

En attendant, reste aussi à faire des revues sur le code du moduile !
@sebastienbeau @bguillot @bealdav @Kev-Roche
